framework/src/libtorch/utils/LibtorchArtificialNeuralNet.C

```diff
@@ -174,6 +185,12 @@ dataStore<Moose::LibtorchArtificialNeuralNet>(
 
   dataStore(stream, items, context);
 
+  int8_t device_type = static_cast<std::underlying_type<torch::DeviceType>::type>(nn->deviceType());
+  dataStore(stream, device_type, context);
+
+  int8_t data_type = static_cast<std::underlying_type<torch::ScalarType>::type>(nn->dataType());
```

Comment:
It seems odd to do a `static_cast` with generic code and then potentially do implicit conversions to `int8_t`. Can you use `auto` instead?

Reply:
Yeah well, this is the reason I used `int8_t`:
```
enum class ScalarType : int8_t {
#define DEFINE_ST_ENUM_VAL_(_1, n) n,
  AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(DEFINE_ST_ENUM_VAL_)
#undef DEFINE_ENUM_ST_ENUM_VAL_
      Undefined,
  NumOptions
};
```
But yeah this should work with out for sure too. I modified it to auto.